csrc/scheduler/resize.cpp

```diff
@@ -313,23 +328,41 @@ void ResizeScheduler::schedule(Fusion* fusion, const HeuristicParams* params) {
   // Make sure the DID ID located at the outermost position
   const auto outermost_pos = scheduler_utils::reorderDevicesToOuter(ref_tv);
 
+  const int64_t vec_factor = resize_params->vectorization_factor;
+
   const int64_t bdimx = 128;
 
-  // Schedule only the remaining IDs
-  ref_tv->flatten(outermost_pos);
-  // [..., I0]
+  int64_t next_innermost_pos = -1;
+  // [..., ...]
+  //        ^
+  //        +--- next_innermost_pos
+
+  if (vec_factor > 1) {
+    ref_tv->split(-1, vec_factor);
+    --next_innermost_pos;
+    // [..., vec_factor]
+    //   ^
+    //   +--- next_innermost_pos
+  }
+
+  ref_tv->flatten(outermost_pos, next_innermost_pos);
+  // [..., I0, vec_factor]
+  //       ^
+  //       +--- next_innermost_pos
 
-  ref_tv->split(-1, bdimx);
-  ref_tv->axis(-1)->parallelize(ParallelType::TIDx);
-  // [..., I0/bdimx, bdimx(TIDx)]
+  ref_tv->split(next_innermost_pos, bdimx);
+  ref_tv->axis(next_innermost_pos)->parallelize(ParallelType::TIDx);
+  --next_innermost_pos;
+  // [..., I0/bdimx, bdimx(TIDx), vec_factor]
+  //         ^
+  //         +--- next_innermost_pos
 
   if (resize_params->split_grid_x_dim) {
-    ref_tv->split(-2, ResizeParams::max_gdimx);
-    // [..., I0/bdimx/max_gdimx, max_gdimx, bdimx(TIDx)]
+    ref_tv->split(next_innermost_pos, ResizeParams::max_gdimx);
+    // [..., I0/bdimx/max_gdimx, max_gdimx, bdimx(TIDx), vec_factor]
   }
-  ref_tv->axis(-2)->parallelize(ParallelType::BIDx);
-  // [..., I0/bdimx/max_gdimx, max_gdimx(BIDx), bdimx(TIDx)] or
-  // [..., I0/bdimx(BIDx), bdimx(TIDx)]
+  ref_tv->axis(next_innermost_pos)->parallelize(ParallelType::BIDx);
+  // [..., I0/bdimx/max_gdimx, max_gdimx(BIDx), bdimx(TIDx), vec_factor]
```

Comment:
did we intend to drop the comment on the possible loop domain from the other code branch?

Reply:
No, I guess it was just dropped while merging PRs.